node/src/integration-test/kotlin/net/corda/node/services/rpc/RpcReconnectTests.kt

```
@@ -0,0 +1,535 @@
+import java.io.InputStream
+import java.io.OutputStream
+import java.net.ServerSocket
+import java.net.Socket
+import java.time.Duration
+import java.util.*
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicReference
+import kotlin.concurrent.thread
+import kotlin.test.assertTrue
+
+/**
+ * This is a slow test.
+ */
+class RpcReconnectTests {
+
+    companion object {
+
+        private val log = contextLogger()
+
+        /**
+         * This function starts the flow returned by [flowStarter] and registers the [onFlowProgressEvent].
+         * In case of connection error it makes a best effort to re-trigger the flow.
+         *
+         * This function attempts to use the [NodeLogin.existingRPCConnection] if provided.
+         *
+         * If the underlying node disconnects this function blocks until the connection is established.
+         *
+         * Currently there is no support in the framework for querying a state of a flow that failed to start or to reconnect to a failed [FlowProgressHandle].
+         *
+         * returns the connection and the [StateMachineRunId] of the started flow - or null if the status of the flow is unknown
+         */
+        fun <T> flowStartWithDisconnectHandling(
+                nodeLogin: NodeLogin,
+                flowStarter: (CordaRPCOps) -> FlowProgressHandle<T>,
+                onFlowProgressEvent: (StateMachineRunId, String) -> Unit
+        ): Pair<CordaRPCConnection, StateMachineRunId?> {
+
+            val connection: CordaRPCConnection = nodeLogin.existingRPCConnection ?: establishConnectionWithRetry(nodeLogin)
+            val rpc = connection.proxy
+
+            fun closeConn() = try {
+                connection.forceClose()
+            } catch (e: Exception) {
```

Can you comment why you're ignoring the exception